pkg/transformer/openshift/openshift.go

```diff
@@ -160,6 +184,10 @@ func (o *OpenShift) Transform(komposeObject kobject.KomposeObject, opt kobject.C
 			objects = append(objects, svc)
 		}
 
+		if service.ExposeService != "" {
+			objects = append(objects, o.initRoute(name, service))
+		}
```

Comment:
Also we can expose a service only if we can create a service in first place. So this if block should be inside `if o.PortsExist(name, service)`.